mmdet/datasets/transforms/loading.py

```diff
@@ -303,49 +303,70 @@ def _poly2mask(self, mask_ann: Union[list, dict], img_h: int,
         mask = maskUtils.decode(rle)
         return mask
 
-    def process_polygons(self, polygons: List[list]) -> List[np.ndarray]:
-        """Convert polygons to list of ndarray and filter invalid polygons.
+    def _process_masks(self, results: dict) -> list:
+        """Process gt_masks and filter invalid polygons.
 
         Args:
-            polygons (list[list]): Polygons of one instance.
+            results (dict): Result dict from :obj:``mmengine.BaseDataset``.
 
         Returns:
-            list[np.ndarray]: Processed polygons.
+            list: Processed gt_masks.
         """
-
-        polygons = [np.array(p) for p in polygons]
-        valid_polygons = []
-        for polygon in polygons:
-            if len(polygon) % 2 == 0 and len(polygon) >= 6:
-                valid_polygons.append(polygon)
-        return valid_polygons
+        gt_masks = []
+        gt_ignore_flags = []
+        for instance in results['instances']:
+            gt_mask = instance['mask']
+            # If the annotation of segmentation mask is invalid,
+            # ignore the whole instance.
+            if isinstance(gt_mask, list):
+                gt_mask = [
+                    np.array(polygon) for polygon in gt_mask
+                    if len(polygon) % 2 == 0 and len(polygon) >= 6
+                ]
+                if len(gt_mask) == 0:
+                    # ignore this instance and set gt_mask to a fake mask
+                    instance['ignore_flag'] = 1
+                    gt_mask = [np.zeros(6)]
+            else:
+                if isinstance(gt_mask, dict) and \
+                        not (gt_mask.get('counts') is not None and
+                             gt_mask.get('size') is not None and
+                             isinstance(gt_mask['counts'], list)):
+                    # ignore this instance and set gt_mask to a fake mask
+                    instance['ignore_flag'] = 1
+                    gt_mask = [np.zeros(6)]
+                if not self.poly2mask:
+                    # deal with invalid polygons
```

Comment:
describe it in detail: `PolygonMasks` requires a ploygon of format `List[np.array]`, other formats are invalid.